### DISCUSSION TITLE
Installing casadi with MATLAB 2022b

### DISCUSSION THREAD

**Discussion by mugheesUCD** [casadi/casadi#3528] [Q&A]
hello,

i need help in installing casadi. I have MATLAB 2022b and i have downloaded and unzipped the "casadi 2018b or later" for the website. 

now when i have unzipped it in C drive, and try to run the following code in matlab, it is showing the following error. 

Unrecognized function or variable 'external'.

Error in PredSim_all_v2 (line 151)
        F = external('F','PredSim_v2.dll');

can please anyone let me know how can i install casadi properly on my computer or guide me where to find the steps to follow for installation.

thanks in advance.
muhammad mughees ahmed

**Comment by jgillis** [comment]
Does the standard check work for you?
```
import casadi.*
x = MX.sym('x')
disp(jacobian(sin(x),x))
```

Does a change from `external` to `casadi.external` work for you?

**Comment by mugheesUCD** [comment]
Hello, 

thank you for your reply.

I have again downloaded the Casadi from website again and unzipped it to be sure about everything. 

I have followed the following steps so far:

1. Unzip casadi files in C drive, My matlab is in C:\Program Files\MATLAB.
      Do i have to move casadi folder into matlab or not?

2. after unzipping the casadi folder that i just wrote following lines in command window 
_import casadi.*
x = MX.sym('x')
disp(jacobian(sin(x),x))_

3. if the casadi folder is selected in the path of matlab then the above mentioned lines run and it shows
_x = 
x
cos(x)_

But if some other path is selected then the following different errors are shown 

_"Unable to resolve the name 'MX.sym'."
"Method 'sym' is not defined for class 'casadi.GenMX' or is removed from MATLAB's search path."_



Do i have to install or unzip all the files of casadi to some specific folder and that is the reason it is not working for me? 

sorry for such a long message, hope it clarifies the problem i am facing. 

Best Regards
Mughees

**Comment by jgillis** [comment]
You'd typically add (without subfolders) the unzipped folder to the matlab path. You can dos this gatphically with a right click, or on the command line with addpath. Moving things into the MATLAB folder does not sound like a good idea.

**Comment by mugheesUCD** [comment]
Thank you for the help!!! and understanding the problem i am facing, it is working fine now.

Thanks and Best regards
Mughees
